radio/bluetooth/bluetooth.go

```diff
@@ -76,20 +79,68 @@ func Disconnect(client ble.Client) error {
 	return nil
 }
 
-func Scan(id string, timeout time.Duration) (map[string]bool, error) {
+func WriteCharacteristic(client ble.Client, characteristic *ble.Characteristic, value []byte, noRsp bool) error {
+	err := make(chan error)
+	go func() {
+		err <- client.WriteCharacteristic(characteristic, value, noRsp)
+	}()
+
+	select {
```

Comment:
Coming back to this, you make a goroutine which runs async and then block anyway. Is there anything stopping you just from running in the same thread, and just returning when done?